xds/internal/balancer/clusterresolver/balancerconfig/configbuilder_test.go

```diff
@@ -0,0 +1,867 @@
+										Weight: 20,
+										ChildPolicy: &internalserviceconfig.BalancerConfig{
+											Name: lrs.Name,
+											Config: &lrs.LBConfig{
+												ClusterName:             testClusterName,
+												EDSServiceName:          testEDSServiceName,
+												LoadReportingServerName: testLRSServer,
+												Locality:                &testLocalityIDs[2],
+												ChildPolicy:             &internalserviceconfig.BalancerConfig{Name: roundrobin.Name},
+											},
+										},
+									},
+									assertString(testLocalityIDs[3].ToString): {
+										Weight: 80,
+										ChildPolicy: &internalserviceconfig.BalancerConfig{
+											Name: lrs.Name,
+											Config: &lrs.LBConfig{
+												ClusterName:             testClusterName,
+												EDSServiceName:          testEDSServiceName,
+												LoadReportingServerName: testLRSServer,
+												Locality:                &testLocalityIDs[3],
+												ChildPolicy:             &internalserviceconfig.BalancerConfig{Name: roundrobin.Name},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				IgnoreReresolutionRequests: true,
+			},
+			"priority-1": {
+				Config: &internalserviceconfig.BalancerConfig{
+					Name: clusterimpl.Name,
+					Config: &clusterimpl.LBConfig{
+						ChildPolicy: &internalserviceconfig.BalancerConfig{Name: "pick_first"},
+					},
+				},
+				IgnoreReresolutionRequests: false,
+			},
+		},
+		Priorities: []string{"priority-0-0", "priority-0-1", "priority-1"},
+	}
+	wantAddrs := []resolver.Address{
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[0][0]}, []string{"priority-0-0", assertString(testLocalityIDs[0].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[0][1]}, []string{"priority-0-0", assertString(testLocalityIDs[0].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[1][0]}, []string{"priority-0-0", assertString(testLocalityIDs[1].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[1][1]}, []string{"priority-0-0", assertString(testLocalityIDs[1].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[2][0]}, []string{"priority-0-1", assertString(testLocalityIDs[2].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[2][1]}, []string{"priority-0-1", assertString(testLocalityIDs[2].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[3][0]}, []string{"priority-0-1", assertString(testLocalityIDs[3].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[3][1]}, []string{"priority-0-1", assertString(testLocalityIDs[3].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[4][0]}, []string{"priority-1"}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[4][1]}, []string{"priority-1"}),
+	}
+
+	if diff := cmp.Diff(gotConfig, wantConfig); diff != "" {
+		t.Errorf("buildClusterImplConfigForDNS() diff (-got +want) %v", diff)
```

Comment:
Shouldn't this be `buildPriorityConfig`?

Reply:
Done